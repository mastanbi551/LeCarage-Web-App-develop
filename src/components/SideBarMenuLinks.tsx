import React from 'react';
import { FiHome, FiUser } from 'react-icons/fi';
import { RiCalendarTodoFill } from 'react-icons/ri';
import {SlCalender} from 'react-icons/sl';

export const SideBarMenuLinks = [
    {
        'path': '/',
        'name': 'Dashboard',
        'icon': <FiHome />,
    },
    {
        'path': '/usertypes',
        'name': 'User Management',
        'icon': <FiUser />,
    },
    {
        'path': '/usertypes',
        'name': 'User Types',
    },
    {
        'path': '/users',
        'name': 'Users',
      },
      {
        'path': '/customers',
        'name': 'Customers',
      },
      {
        'path': '/cars',
        'name': 'Cars',
      },
      {
        'path': '/BookingsList',
        'name': 'Bookings',
        'icon': <RiCalendarTodoFill />,
      },
      {
        'path': '/BookingsList',
        'name': 'All Bookings',
        // icon: <FaMoneyBill />,
      },
      {
        'path': '/Calender',
        'name': 'Calendar',
        'icon': <SlCalender />,
      },
      {
        'path': '/Calender',
        'name': 'Events'
      },
      {
        'path': 'Services',
        'name': 'Services'
      }
];

export const DefaultMenu = [
    {
        'path': '/',
        'name': 'Dashboard',
        'icon': <FiHome />,
    },
];