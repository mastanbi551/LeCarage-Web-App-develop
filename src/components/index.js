import { SideBar } from './sidebar/SideBar';


export { CustomButton } from './customButton/CustomButton';
export { BackgroundImageContainer } from './backgroundImageContainer/BackgroundImageContainer';
export { LeCarageLogo } from './leCarageLogo/LeCarageLogo';
export { PageHeader } from './pageHeader/PageHeader';
export { InputTextField } from './inputTextField/InputTextField';
export { Header } from './header/Header';
export { CustomDropDown } from './customDropDown/CustomDropDown';
export { CheckBoxList } from './checkBoxList/CheckBoxList';
export { Loader } from './loader/Loader';

export { PageContainer } from './pageContainer/PageContainer';
export { DetailsContainer } from './pageContainer/DetailsContainer';
export { CardContainer } from './pageContainer/CardContainer';


export { DynamicDataTable } from './dynamicDataTable/DynamicDataTable';
export { DynamicDataTableHeader } from './dynamicDataTable/DynamicDataTableHeader';
export { DynamicDataTableItems } from './dynamicDataTable/DynamicDataTableItems';
export { CustomRadioButtonList } from './customRadioButtonsList/CustomRadioButtonsList';
export { CustomFilter } from './customFilter/CustomFilter';

export { GooglePlaces } from '../components/googlePlaces/GooglePlaces';
export { CustomCheckBox } from '../components/customCheckBox/CustomCheckBox';
export { SearchableDropdown } from './searchableDropDown/SearchableDropdown';

export default SideBar;

