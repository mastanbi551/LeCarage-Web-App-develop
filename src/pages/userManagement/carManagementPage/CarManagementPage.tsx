import React from 'react';
import {
  DynamicDataTable,
  PageHeader,
} from '../../../components';
import './CarManagementPage.scss';

export interface CarProps { 
  carId: string,
  Color: string,
  Customer: string, 
  Location: string,
  Make: string,
  Model: string,
  'Number Plate': string,
  Stalling: string,
  Status: string,
}
interface CarManagementPageInterface {
  handleAddCar: () => void;
    columns: { Header: string; accessor: string; }[];
    searchValue:  string[],
    onViewHandler: (rowInfo: object) => void;
    onEditHandler: (rowInfo: object) => void;
    onDeleteHandler: (rowInfo: object) => void;
    carsInfo: [];
    errorMessage: string;
    filterValues: {mainMenu: string, subMenus: string[]}[];
    access: string;
    rowsPerPageValueChange: (value: string) => void;  
    itemsPerPage: number
}
export const CarManagementPage: React.FC<CarManagementPageInterface> = ({
  handleAddCar,
  columns,
  searchValue,
  onViewHandler,
  onEditHandler,
  onDeleteHandler,
  carsInfo,
  errorMessage,
  filterValues,
  access,
  rowsPerPageValueChange,
  itemsPerPage
}) => {


  return (
    <>
    
      <div className='cars-container'>
        <div className='cars-container__header-content'>
          <PageHeader
            text='Cars'
            name='cars-container__header-content__heading'
            isButtonDisable={true}
            buttonName='Add New'
            buttonClassName='cars-container__header-content__button greenButton btn btn-success'
            buttonOnclick={handleAddCar}
            access={access}
          />
          
        </div>
        

        <div className='cars-container__table-content '>
          <div className='cars-container__table'>
            <DynamicDataTable
              tableHeading=''
              isSearchIconEnable={true}
              isFilterIconEnable={true}
              columns={columns}
              data={carsInfo}
              searchValue={searchValue}              
              filterValues={filterValues}
              onViewHandler={onViewHandler}
              onEditHandler={onEditHandler}
              onDeleteHandler={onDeleteHandler}
              defaultItemsPerPage={itemsPerPage}
              errorMessage={errorMessage}
              isEditIconEnable={true}
              isDeleteIconEnable={true}
              access={access}
              rowsPerPageValueChange={rowsPerPageValueChange}
            />
          </div>
        </div>
      </div>
    </>
  );

};
