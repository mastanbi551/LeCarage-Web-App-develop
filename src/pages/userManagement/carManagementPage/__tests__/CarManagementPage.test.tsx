import React from 'react';
import { render, screen } from '@testing-library/react';
import { DynamicDataTable, PageHeader } from '../../../../components';
import { CarManagementPage } from '../CarManagementPage';

describe('CarManagementPage', () => {
  const carsInfo = [
    {
      carId: '1',
      Color: 'Blue',
      Customer: 'John Doe',
      Location: 'New York',
      Make: 'Honda',
      Model: 'Civic',
      'Number Plate': 'ABC123',
      Stalling: 'No',
      Status: 'Active',
    },
    // Add more cars as needed for testing
  ];
  const columns = [
    { Header: 'Car ID', accessor: 'carId' },
    { Header: 'Color', accessor: 'Color' },
    // Add more columns as needed for testing
  ];
  const searchValue = ['Honda'];
  const filterValues = [
    { mainMenu: 'Status', subMenus: ['Active', 'Inactive'] },
    // Add more filter values as needed for testing
  ];
  const errorMessage = '';

  const mockHandleAddCar = jest.fn();
  const mockOnViewHandler = jest.fn();
  const mockOnEditHandler = jest.fn();
  const mockOnDeleteHandler = jest.fn();

  it('renders correctly', () => {
    const { container } = render(
      <CarManagementPage
        handleAddCar={mockHandleAddCar}
        columns={columns}
        searchValue={searchValue}
        filterValues={filterValues}
        carsInfo={carsInfo}
        errorMessage={errorMessage}
        onViewHandler={mockOnViewHandler}
        onEditHandler={mockOnEditHandler}
        onDeleteHandler={mockOnDeleteHandler}
      />
    );

    expect(container).toMatchSnapshot();
  });

  it('should call handleAddCar when Add New button is clicked', () => {
    render(
        <CarManagementPage
        handleAddCar={mockHandleAddCar}
        columns={columns}
        searchValue={searchValue}
        filterValues={filterValues}
        carsInfo={carsInfo}
        errorMessage={errorMessage}
        onViewHandler={mockOnViewHandler}
        onEditHandler={mockOnEditHandler}
        onDeleteHandler={mockOnDeleteHandler}
      />
    );

    const addButton = screen.getByRole('button', { name: 'Add New' });
    addButton.click();

    expect(mockHandleAddCar).toHaveBeenCalled();
  });

});
