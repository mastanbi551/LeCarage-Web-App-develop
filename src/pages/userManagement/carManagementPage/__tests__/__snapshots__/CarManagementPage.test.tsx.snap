// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CarManagementPage renders correctly 1`] = `
<div>
  <div
    class="cars-container"
  >
    <div
      class="cars-container__header-content"
    >
      <h1
        class="cars-container__header-content__heading"
      >
        Cars
      </h1>
      <button
        class="cars-container__header-content__button greenButton btn btn-success"
        type="button"
      >
        Add New
      </button>
    </div>
    <div
      class="cars-container__table-content "
    >
      <div
        class="cars-container__table"
      >
        <div
          class="loader"
        >
          <div
            aria-busy="true"
            aria-label="three-dots-loading"
            data-testid="three-dots-loading"
            role="status"
            style="display: flex;"
          >
            <svg
              data-testid="three-dots-svg"
              fill="#4fa94d"
              height="80"
              viewBox="0 0 120 30"
              width="80"
              xmlns="http://www.w3.org/2000/svg"
            >
              <circle
                cx="15"
                cy="15"
                r="15"
              >
                <animate
                  attributeName="r"
                  begin="0s"
                  calcMode="linear"
                  dur="0.8s"
                  from="15"
                  repeatCount="indefinite"
                  to="15"
                  values="15;9;15"
                />
                <animate
                  attributeName="fill-opacity"
                  begin="0s"
                  calcMode="linear"
                  dur="0.8s"
                  from="1"
                  repeatCount="indefinite"
                  to="1"
                  values="1;.5;1"
                />
              </circle>
              <circle
                attributeName="fill-opacity"
                cx="60"
                cy="15"
                from="1"
                r="9"
                to="0.3"
              >
                <animate
                  attributeName="r"
                  begin="0s"
                  calcMode="linear"
                  dur="0.8s"
                  from="9"
                  repeatCount="indefinite"
                  to="9"
                  values="9;15;9"
                />
                <animate
                  attributeName="fill-opacity"
                  begin="0s"
                  calcMode="linear"
                  dur="0.8s"
                  from="0.5"
                  repeatCount="indefinite"
                  to="0.5"
                  values=".5;1;.5"
                />
              </circle>
              <circle
                cx="105"
                cy="15"
                r="15"
              >
                <animate
                  attributeName="r"
                  begin="0s"
                  calcMode="linear"
                  dur="0.8s"
                  from="15"
                  repeatCount="indefinite"
                  to="15"
                  values="15;9;15"
                />
                <animate
                  attributeName="fill-opacity"
                  begin="0s"
                  calcMode="linear"
                  dur="0.8s"
                  from="1"
                  repeatCount="indefinite"
                  to="1"
                  values="1;.5;1"
                />
              </circle>
            </svg>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
